### DISCUSSION TITLE
Does Mingingcore runing well in Ropsten Network？My PPLNS payment process didnt work

### DISCUSSION THREAD

**Discussion by James1847** [coinfoundry/miningcore#1103] [Q&A]
Looks like mingcore runs well with the logs below:
![image](https://user-images.githubusercontent.com/29943599/149607279-ed2e43c0-eda3-4df5-adad-00e514d3eea4.png)
But in reality, no eth rewards received from the ETH pool account

Jan 15 03:38:21 ip-172-31-42-4 dotnet[28825]: [2022-01-15 03:38:21.0906] [I] [PayoutManager] Processing payments for pool eth1, block 11817044
Jan 15 03:38:21 ip-172-31-42-4 dotnet[28825]: [2022-01-15 03:38:21.0948] [I] [PayoutManager] Processing payments for pool eth1, block 11817047
Jan 15 03:38:21 ip-172-31-42-4 dotnet[28825]: [2022-01-15 03:38:21.0977] [I] [eth1] [Ethereum Payout Handler] Sending 2.03648 ETH to 0x02ecfcf0aa792b1e2ad1987a9c0e2752ef14ea32
Jan 15 03:38:21 ip-172-31-42-4 dotnet[28825]: [2022-01-15 03:38:21.1018] [I] [eth1] [Ethereum Payout Handler] Payment transaction id: 0x02b55b1bed37a42340c0eac442b9ae24c751e7fc4c0fe13e5c42dde772b6fb61

**Comment by ei8ht187** [comment]
It has some hiccups. Sometimes, shares are doubled written down in the database. As it is the testnet, I just deleted this shares manually from the database. There is also an old issue #350 discussing the same matter.

Maybe, or special for production, there is a better way. But I'm not a DBA. Maybe @oliverw can help out in that question.

**Reply by oliverw** [reply]
> It has some hiccups. Sometimes, shares are doubled written down in the database

I have a really hard time believing this. Shares are written inside an (ACID) transaction.

**Reply by ei8ht187** [reply]
Just have the issue now. I mixed up shares and blocks. This is the error from the database: 
 2022-01-17T11:51:51+01:00 postgresql-14.service [23-1] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] ERROR: duplicate key value violates unique constraint "blocks_pool_height"

2022-01-17T11:51:51+01:00 postgresql-14.service [23-3] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] STATEMENT: COMMIT

2022-01-17T11:51:51+01:00 postgresql-14.service [23-2] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] DETAIL: Key (poolid, blockheight, type)=(eth1, 11830661, block) already exists. 

My solution to solve this issue: search the whole database for "11830661" an delete every entry. On Ropsten I don't care about. But in prod I need another solution in case that happens

**Reply by oliverw** [reply]
> Just have the issue now. I mixed up shares and blocks. This is the error from the database: 2022-01-17T11:51:51+01:00 postgresql-14.service [23-1] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] ERROR: duplicate key value violates unique constraint "blocks_pool_height"
> 
> 2022-01-17T11:51:51+01:00 postgresql-14.service [23-3] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] STATEMENT: COMMIT
> 
> 2022-01-17T11:51:51+01:00 postgresql-14.service [23-2] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] DETAIL: Key (poolid, blockheight, type)=(eth1, 11830661, block) already exists.
> 
> My solution to solve this issue: search the whole database for "11830661" an delete every entry. On Ropsten I don't care about. But in prod I need another solution in case that happens

Hmm, how can there be a duplicate block in the `blocks` table if the insert failed in the first place due to `duplicate key value violates unique constraint "blocks_pool_height"` ?

**Reply by ei8ht187** [reply]
The logs from miningcore shows the same block twice, log below is in UTC. Probably because we have two stratum proxies and one masternode and at the moment aboug 90% of hashpower in ropsten:

{InstantSeconds=1642415547},ISO,Z resolved to 2022-01-17T10:32:27.384	masternode	[2022-01-17 10:32:27.3838] [I] [eth1] [Ethereum Payout Handler] Unlocked block 11830661 worth 2 ETH
{InstantSeconds=1642415547},ISO,Z resolved to 2022-01-17T10:32:27.387	masternode	[2022-01-17 10:32:27.3838] [I] [eth1] [Ethereum Payout Handler] Unlocked block 11830661 worth 2 ETH

From the database I have not much more in term of logs. For the masternode I can provide you a full log via mail. 

Kind regards

ei8ht

**Reply by James1847** [reply]
Hope u guys can solve this problems, but now I have to test in mainnet, lol

**Reply by BitWizJason** [reply]
I have also experienced the duplicate block issue within the blocks table while on the Ropsten testnet. I updated that rows status column to some outlandish value like 99 so that the PayoutManager would stop trying to process it and throw exceptions. I was unable to identify when and how the duplicate got inserted.

**Reply by ei8ht187** [reply]
> > Just have the issue now. I mixed up shares and blocks. This is the error from the database: 2022-01-17T11:51:51+01:00 postgresql-14.service [23-1] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] ERROR: duplicate key value violates unique constraint "blocks_pool_height"
> > 2022-01-17T11:51:51+01:00 postgresql-14.service [23-3] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] STATEMENT: COMMIT
> > 2022-01-17T11:51:51+01:00 postgresql-14.service [23-2] pid=302203,user=miningcore,db=miningcore,app=[unknown],client=[IP PRIMARYNODE] DETAIL: Key (poolid, blockheight, type)=(eth1, 11830661, block) already exists.
> > My solution to solve this issue: search the whole database for "11830661" an delete every entry. On Ropsten I don't care about. But in prod I need another solution in case that happens
> 
> Hmm, how can there be a duplicate block in the `blocks` table if the insert failed in the first place due to `duplicate key value violates unique constraint "blocks_pool_height"` ?

Can confirm: there are duplicated block entries in the block table. One in status completed, one in status pending. Deleting the compleded entry solves the problem. I'm just hoping this won't happen in production. Is this is the correct fix for this problem, just in case it will happen in production?

DB before edit:
![Bildschirmfoto vom 2022-02-06 20-53-56](https://user-images.githubusercontent.com/72598876/152699465-aac0354a-d075-40d0-902d-c661ec9ee5f1.png)

DB after edit: 
![Bildschirmfoto vom 2022-02-06 20-55-55](https://user-images.githubusercontent.com/72598876/152699474-a27ef111-f6cf-4168-95e5-649d1a8fa990.png)

**Reply by oliverw** [reply]
Yes, it is. Do you we have an issue for this yet? If not please create one.

**Reply by ei8ht187** [reply]
@oliverw there was issue #1080 but it's already closed because no one delivered more informations till yesterday. 
Do you reopen this one or should I create a new one?
